src/plugins/data_view_editor/public/data_view_editor_service.ts

```diff
@@ -17,30 +25,117 @@ import {
 import { GetFieldsOptions } from './shared_imports';
 
+export const matchedIndiciesDefault = {
+  allIndices: [],
+  exactMatchedIndices: [],
+  partialMatchedIndices: [],
+  visibleIndices: [],
+};
+
+export interface DataViewEditorServiceConstructorArgs {
+  services: {
+    http: HttpSetup;
+    dataViews: DataViewsServicePublic;
+  };
+  requireTimestampField?: boolean;
+  initialValues: {
+    name?: string;
+    type?: INDEX_PATTERN_TYPE;
+    indexPattern?: string;
+  };
+}
+
 export class DataViewEditorService {
-  constructor(private http: HttpSetup, private dataViews: DataViewsServicePublic) {
+  constructor({
+    services: { http, dataViews },
+    initialValues: {
+      type: initialType = INDEX_PATTERN_TYPE.DEFAULT,
+      indexPattern: initialIndexPattern = '',
+      name: initialName = '',
+    },
+    requireTimestampField = false,
+  }: DataViewEditorServiceConstructorArgs) {
+    this.http = http;
+    this.dataViews = dataViews;
+    this.requireTimestampField = requireTimestampField;
+    this.type = initialType;
+    this.indexPattern = removeSpaces(initialIndexPattern);
+
+    // fire off a couple of requests that we know we'll need
     this.rollupCapsResponse = this.getRollupIndexCaps();
+    this.dataViewNames$ = from(this.loadDataViewNames(initialName));
+
+    // public observables
+    this.matchedIndices$ = this.matchedIndicesInternal$.asObservable();
+    this.rollupIndicesCaps$ = this.rollupIndicesCapsInternal$.asObservable();
+    this.isLoadingSources$ = this.isLoadingSourcesInternal$.asObservable();
+    this.loadingTimestampFields$ = this.loadingTimestampFieldsInternal$.asObservable();
+    this.timestampFieldOptions$ = this.timestampFieldOptionsInternal$.asObservable();
+    this.rollupIndex$ = this.rollupIndexInternal$.asObservable();
+
+    // when list of matched indices is updated always update timestamp fields
+    this.loadTimestampFieldsSub = this.matchedIndices$.subscribe(() => this.loadTimestampFields());
+
+    // alternate value with undefined so validation knows when its getting a fresh value
+    this.matchedIndicesForProviderSub = this.matchedIndices$.subscribe((matchedIndices) => {
+      this.matchedIndicesForProvider$.next(matchedIndices);
+      this.matchedIndicesForProvider$.next(undefined);
+    });
+
+    // alternate value with undefined so validation knows when its getting a fresh value
+    this.rollupIndexForProviderSub = this.rollupIndex$.subscribe((rollupIndex) => {
+      this.rollupIndexForProvider$.next(rollupIndex);
+      this.rollupIndexForProvider$.next(undefined);
+    });
   }
 
-  rollupIndicesCapabilities$ = new BehaviorSubject<RollupIndicesCapsResponse>({});
-  isLoadingSources$ = new BehaviorSubject<boolean>(false);
+  private http: HttpSetup;
+  private dataViews: DataViewsServicePublic;
+  // config
+  private requireTimestampField: boolean;
+  private type = INDEX_PATTERN_TYPE.DEFAULT;
+
+  // state
+  private indexPattern = '';
+  private allowHidden = false;
```

Comment:
Would be great if `allowHidden` becomes enabled when editing a saved data view for hidden indices.

Steps:
- Create a data view for hidden indices
- Open it in Edit mode and see that it can't find matching indices unless the switch is toggled manually again.

Reply:
Also it does not restore a saved time field value:

![Nov-28-2022 10-32-37](https://user-images.githubusercontent.com/1415710/204243146-bc3181bf-6277-496c-bb4e-81ca545f8aaf.gif)

Reply:
I'll address these in a follow up PR.